app/dictionaries/language_dictionary.rb

```diff
@@ -1,39 +1,41 @@
 # Dictionary of Languages
 class LanguageDictionary < Dictionary
 
-  DEFAULT_FILE = 'language.yml'
+  # Not really a dictionary (an array) ...
+  def initialize
+    @dictionary = TeSS::Config.languages
+  end
+
+  def lookup(id)
+    @dictionary.find { |v| v == id }
+  end
 
   # Override because we want this in the viewers language ...
   def options_for_select(existing = nil)
     d = if existing
-          @dictionary.select { |key, _value| existing.include?(key) }
+          @dictionary.select { |key| existing.include?(key) }
         else
           @dictionary
         end
 
-    d.map do |key, _value|
-      [render_language_name(key), key, key]
-    end
+    d.map { |key| [render_language_name(key), key, key] }
   end
 
   def values_for_search(keys)
     return unless keys
-    @dictionary.select { |key, _value| keys.include?(key) }.
-      map { |key, _value| render_language_name(key) }
+    @dictionary.select { |key| keys.include?(key) }.
+      map { |key| render_language_name(key) }
   end
 
   def render_language_name(code)
-    # I18nData lookup wants uppercase key
-    # TODO: is capitalized a good choice for all locales?
-    i18ndata_code = code.to_s.upcase
-    I18nData.languages(I18n.locale)[i18ndata_code].capitalize
-  end
-
-  private
-
-  def dictionary_filepath
-    get_file_path 'language', DEFAULT_FILE
+    begin
+      I18n.t("languages.#{code.downcase}", raise: true)
```

Comment:
You can supply a `default: "..."` argument to `I18n.t` to be used as the fallback if the key is missing.

Reply:
I'll give it a try, but I think this might not pass in the test suite because of:

https://github.com/ElixirTeSS/TeSS/blob/master/config/environments/test.rb#L47

Reply:
Nope seemed to work fine.
I pushed this in the latest changes, also added some dictionary unit tests.